fdbclient/ServerKnobs.cpp

```diff
@@ -481,6 +481,9 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 	init( RESET_MASTER_DELAY,                                   300.0 );
 	init( RESET_RESOLVER_DELAY,                                 300.0 );
 
+	init( GLOBAL_CONFIG_MIGRATE_TIMEOUT,                          1.0 );
```

Comment:
I added a version of this, essentially starting the refresh in parallel with the migration, and not serving requests until either the migration finished or timed out. I think the danger of allowing the migration to keep running after serving requests is that a client could read a value that the migration then updates. I think these scenarios are pretty unlikely, since it requires a specific key to be read right after starting a database that has just been migrated from an older version that also has that key set.

Reply:
> I think the danger of allowing the migration to keep running after serving requests is that a client could read a value that the migration then updates

Is this a risk in the timeout case as well? For example, if one proxy times out and then another proxy completes the migrate.

Reply:
Probably yeah. Not sure how much work it's worth putting in to fix it though. I guess the easiest solution would be to only run the migration on one proxy, although I'm not sure of a deterministic way to determine that.

Reply:
The issue is the logic is custom in that it reads a specific, old `\xff` key. The update would also have a similar problem to above where it could be called from multiple clients.

Reply:
Is the update done via a transaction, or does it happen through some other mechanism? If it is transactional, then you could do something like read and clear the old keys and update the config in a single transaction. Then only one proxy could successfully read and clear the old keys, and the others would get a conflict or see that the keys are already gone.

Reply:
Actually we may be ok then. It is updated through a transaction, and sets `migratedKey` to a true value during the transaction. Each attempt also reads `migratedKey` and does nothing if it is set.